backend/utils.py

```
@@ -71,21 +79,33 @@ def _validate_uploaded_file_sync(file: UploadFile) -> Optional[Image.Image]:
+            try:
+                # Read first 1024 bytes for MIME detection
+                file_content = file.file.read(1024)
+                file.file.seek(0)  # Reset file pointer
+
+                detected_mime = magic.from_buffer(file_content, mime=True)
+
+                if detected_mime not in ALLOWED_MIME_TYPES:
+                    raise HTTPException(
+                        status_code=400,
+                        detail=f"Invalid file type. Only image files are allowed. Detected: {detected_mime}"
+                    )
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)